diff --git a/node_modules/proxy-from-env/index.js b/node_modules/proxy-from-env/index.js
index df75004..056b8b6 100644
--- a/node_modules/proxy-from-env/index.js
+++ b/node_modules/proxy-from-env/index.js
@@ -1,6 +1,6 @@
 'use strict';
 
-var parseUrl = require('url').parse;
+var URL = require('url').URL;
 
 var DEFAULT_PORTS = {
   ftp: 21,
@@ -22,7 +22,16 @@ var stringEndsWith = String.prototype.endsWith || function(s) {
  *  given URL. If no proxy is set, this will be an empty string.
  */
 function getProxyForUrl(url) {
-  var parsedUrl = typeof url === 'string' ? parseUrl(url) : url || {};
+  var parsedUrl;
+  if (typeof url === 'string') {
+    try {
+      parsedUrl = new URL(url);
+    } catch (e) {
+      parsedUrl = {};
+    }
+  } else {
+    parsedUrl = url || {};
+  }
   var proto = parsedUrl.protocol;
   var hostname = parsedUrl.host;
   var port = parsedUrl.port;
